filebeat/fileset/pipelines.go

```diff
@@ -129,6 +135,37 @@ func loadPipeline(esClient PipelineLoader, pipelineID string, content map[string
 	return nil
 }
 
+// setEcsProcessors sets required ECS options in processors when filebeat version is 7.0.X
+// and ES is 6.7.X to ease migration to ECS
+func setEcsProcessors(esVersion common.Version, pipelineID string, content map[string]interface{}) error {
+	ecsVersion := common.MustNewVersion("7.0.0")
+	if !esVersion.LessThan(ecsVersion) {
+		return nil
+	}
+
+	processors, ok := content["processors"].([]interface{})
+	if !ok {
+		logp.Debug("modules", "Pipeline '%s' without processors: %+v", pipelineID, content)
+		return nil
+	}
+
+	minUserAgentVersion := common.MustNewVersion("6.7.0")
+	for _, p := range processors {
+		processor, ok := p.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		if options, ok := processor["user_agent"].(map[string]interface{}); ok {
+			if esVersion.LessThan(minUserAgentVersion) {
+				return fmt.Errorf("user_agent processor requires option `ecs: true`, available in Elasticsearch %v", minUserAgentVersion)
```

Comment:
I propose including the user's ES version like `fmt.Errorf("user_agent processor requires option 'ecs: true', but Elasticsearch %v does not support this option (Elasticsearch %v or newer is required)", esVersion, minUserAgentVersion)`.